docs/Gamification_extension_mechanism.rst

```diff
@@ -0,0 +1,175 @@
+                    <value>Development</value>
+                </value-param>
+                <value-param>
+                    <name>zone-priority</name>
+                    <value>5</value>
+                </value-param>
+            </init-params>
+        </component-plugin>
+    </external-component-plugins>
+
+===============
+Add new badges
+===============
+
+To add new badges to the gamification configuration, a simple external component plugin should be added to your addon configuration. This plugin should define the following information:
+
+Badge configuration
+~~~~~~~~~~~~~~~~~~~
+-  ``badge-title``: The name of the new badge.
+-  ``badge-description``: The description of the new badge.
+-  ``badge-domain``: The domain of the new badge.
+-  ``badge-icon``: The icon of the badge (ex: icon_name.png), This icon file should be added under the folder '\resources\medias\images' of your addon’s jar.
+-  ``badge-neededScore``: The score required in the domain to earn this badge.
+-  ``badge-enable``: Define if the badge is enabled or not (true/false)
+
+
+
+.. code:: xml
+
+    <external-component-plugins>
+        <target-component>org.exoplatform.addons.gamification.service.setting.badge.BadgeRegistry</target-component>
+        <component-plugin>
+            <name>badge.SocialNovice</name>
+            <set-method>addPlugin</set-method>
+            <type>org.exoplatform.addons.gamification.service.setting.badge.model.BadgeConfig</type>
+            <init-params>
+                <value-param>
+                    <name>badge-title</name>
+                    <value>Social Novice</value>
+                </value-param>
+                <value-param>
+                    <name>badge-description</name>
+                    <value>Growing your network and ramping up your interactions</value>
+                </value-param>
+                <value-param>
+                    <name>badge-domain</name>
+                    <value>Social</value>
+                </value-param>
+                <value-param>
+                    <name>badge-icon</name>
+                    <value>socialNovice.png</value>
+                </value-param>
+                <value-param>
+                    <name>badge-neededScore</name>
+                    <value>50</value>
+                </value-param>
+                <value-param>
+                    <name>badge-enable</name>
+                    <value>true</value>
+                </value-param>
+            </init-params>
+        </component-plugin>
+    </external-component-plugins>
+
+
+===============
+Add new event
+===============
+
+Gamification uses eXo Event/Listener mechanism to grant points when configured events match a rule. To trigger a rule in the gamification system all you have to do is to broadcast a specific gamification event from your addon code using the eXo ListenerService.
+This can be done by adding the following code:
+
+
+.. code:: java
+
```

Comment:
should shoulw the import here so that developer does not have to look for it. We may also indicate which jar (maven dependency) they should add. Or maybe this tutorial should simply start by giving a link to the GitHub connector code repo as a sample code.